### ISSUE TITLE
sysytemd-journal-flush.service depends on systemd-user-sessions.service

### ISSUE THREAD

**Issue by pasqui23** [systemd/systemd#10469]
**systemd version the issue has been seen with**
>
```
 systemd 239
+PAM +AUDIT -SELINUX +IMA +APPARMOR +SMACK -SYSVINIT +UTMP -LIBCRYPTSETUP +GCRYPT -GNUTLS +ACL +XZ +LZ4 +SECCOMP +BLKID -ELFUTILS +KMOD +IDN2 -IDN +PCRE2 default-hierarchy=hybrid
```

**Used distribution**
 > nixos unstable

**Expected behaviour you didn't see**
 > Fast startup
**Unexpected behaviour you saw**
 > sysytemd-journal-flush.service appearing in the critical chain as a dependency of systemd-user-sessions.service

**Steps to reproduce the problem**

**Comment by bl33pbl0p** [comment]
Not sure what looks wrong to you, the journal is flushed to disk and writes made persistent (if configured) so that when users are allowed to login their logs actually go to disk, journal-flush happens before that, and sounds correct to me.

**Comment by pasqui23** [comment]
It blocks `systemd-user-sessions` and can add even 1 minute to startup.
`systemd-journal-flush` can't happen in background,without blocking other services?

**Comment by poettering** [comment]
Hmm, I wonder why we added the ordering dep in the first place. It has been there since the file was first committed. I don't really see why we have it now, maybe we can just drop it.

**Comment by poettering** [comment]
Fix waiting in #10502
